### PR TITLE
Soleng 1959 add mime receipt message details to 0x23

### PR DESCRIPTION
**Pull request by muyuncaigeotab**

### CODE REVIEW

**Comment by Stephen-F** [comment]
QAed the content change in rs232-usb.md. I have no knowledge of 'Gemfile' so I can't QA the changes to the other 2 files.

File: src/hardware/addon-protocols/rs232-usb.md [hunk 1]
```diff
@@ -379,13 +379,18 @@ Third-Party Data Acknowledge from GO device
 0x02, 0x02, 0x00, 0x04, 0x0A, 0x03
 ```
 
-#### Appendix B: Sample Message Flow for IOX-USB & IOX-RS232
+#### <a name="appendixB"></a> Appendix B: Sample Message Flow for IOX-USB & IOX-RS232
```
Comment (beefo) [inline src/hardware/addon-protocols/rs232-usb.md:382]: Can you see if [this](https://stackoverflow.com/questions/27981247/github-markdown-same-page-link) works as a substitute to using anchor tags in the mardown?

File: .gitignore [hunk 2]
```diff
@@ -67,3 +67,6 @@ dist
 # hide the AWS and Rsync credentials file
 aws-credentials.json
 rsync-credentials.json
+
+# ignore the result of running Gemfile
+Gemfile.lock
```
[the remark this answers is not in the document]
Comment (muyuncaigeotab) [reply .gitignore:72]: I misunderstood what Gemfile.lock does, I will revert the change.

File: src/hardware/addon-protocols/rs232-usb.md (hunk 1 above)
Comment (muyuncaigeotab) [reply src/hardware/addon-protocols/rs232-usb.md:382]: Yes, it works. I can change to that.

**Pull request merged**